ert_shared/ensemble_evaluator/prefect_ensemble/function_step.py

```diff
@@ -0,0 +1,93 @@
+from prefect import Task
+from typing import Callable
+from ert_shared.ensemble_evaluator.prefect_ensemble.client import Client
+from ert_shared.ensemble_evaluator.entity import identifiers as ids
+from ert_shared.ensemble_evaluator.prefect_ensemble.storage_driver import (
+    storage_driver_factory,
+)
+
+
+class FunctionStep(Task):
+    def __init__(
+        self,
+        step,
+        url,
+        ee_id,
+        storage_config,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self._step = step
+        self._url = url
+        self._ee_id = ee_id
+        self._storage = storage_driver_factory(storage_config, ".")
+
+    def get_iens(self):
+        return self._step["iens"]
+
+    def get_stage_id(self):
+        return self._step["stage_id"]
+
+    def get_step_id(self):
+        return self._step["step_id"]
+
+    def get_ee_id(self):
+        return self._ee_id
+
+    @property
+    def job(self):
+        return self._step["jobs"][0]
+
+    @property
+    def step_source(self):
+        iens = self.get_iens()
+        stage_id = self.get_stage_id()
+        step_id = self.get_step_id()
+        return f"/ert/ee/{self._ee_id}/real/{iens}/stage/{stage_id}/step/{step_id}"
+
+    def function_source(self, fun_id):
+        return f"{self.step_source}/job/{fun_id}"
+
+    def run_job(self, client, job):
+        try:
+            result = job["executable"](**self._step["step_input"])
```

Comment:
Are the step inputs supposed to be only literal values, or should they also be able to come from other tasks via storage?

Reply:
They should be able to come from tasks as well via storage, but that is currently not really well defined and to just get things started I assume only literal values for now.